src/app/core/config/config.service.ts

```diff
@@ -68,6 +68,7 @@ export class ConfigService extends LatestEntityLoader<Config> {
       migrateEntityBlock,
       migrateGroupByConfig,
       addDefaultNoteDetailsConfig,
+      migratePublicFormConfig,
```

Comment:
This will probably not work yet because these migrations are only applied to the "big" Config:CONFIG_ENTITY doc and not to other Config entities ...

maybe we should move this migration code to the PublicForm component and execute it there?

Reply:
As we removed the `PublicFormConfig.prefilled` from the PublicConfig and its usages from component file, so still we need to use this migration in PublicComponent?

Reply:
There migration will be needed for backward compatibility. Some clients system config will still have this old format. But let's finish the basic implementation and then think how the migration would look

Reply:
This migrations are just handling the Config:CONFIG_ENTITY document. For this kind of migration we need here, is currently no approach implemented.

What we should do, is to do this migration directly in the public-form.component. To be able to test it, you can use this real config from another instance:

```json
{
  "_id": "PublicFormConfig:1",
  "_rev": "2-d0073d2edeb0f1bf419c299603263dba",
  "title": "Mentee Test Formular",
  "description": "Dies ist nur ein Beispiel Formular für Mentees. Interessierte können dieses Formular ausfüllen und die Daten werden danach durch die Mitarbeiter:innen eingesehen.",
  "entity": "Child",
  "columns": [
    [
      "vorname"
    ],
    [
      "name"
    ],
    [
      "gender"
    ],
    [
      "dateOfBirth"
    ],
    [
      "email"
    ],
    [
      "phone"
    ]
  ],
  "prefilled": {
    "projectStatus": "AUSWAHL"
  }
}
```

Reply:
@Abhinegi2 , I have added a unit test for migration because we also need support to migrate from the old "prefilled" config field (this seems to not work yet - neither in the app when I test on an old public form config stored in my db, nor in the new unit test).

Have you implemented migration for the default values that are defined as "prefilled" yet? Can you please look into this?

Reply:
When testing I still notice some problems even though the unit test passed (not sure what I missed in the test case). I have adjusted the logic myself a bit and got it working now.